opm/simulators/wells/GroupState.cpp

```diff
@@ -419,6 +423,63 @@ has_gpmaint_target(const std::string& gname) const
     return (this->m_gpmaint_target.count(gname) > 0);
 }
 
+template<class Scalar>
+void GroupState<Scalar>::
+update_gconsump(const Schedule& schedule, const int report_step, const SummaryState& summary_state) {
+    this->m_gconsump_rates.clear();
+    const auto& sched_state = schedule[report_step];
+    const auto& gconsump = sched_state.gconsump();
+    auto gcr_recursive =
+        [this, &sched_state, &gconsump, &summary_state](auto self,
+                                                        const std::string& group_name,
+                                                        std::pair<Scalar, Scalar>& rates,
+                                                        const double parent_gefac = 1.0) -> bool
+        {
+            // If group already has been computed, update parent rates and return true
+            const auto it = this->m_gconsump_rates.find(group_name);
+            if (it != this->m_gconsump_rates.end()) {
+                rates.first += static_cast<Scalar>(it->second.first * parent_gefac);
+                rates.second += static_cast<Scalar>(it->second.second * parent_gefac);
+                return true;
+            }
+
+            // Accumulate from sub-groups and keep track of any updates in 'has_values'
+            bool has_values = false;
+            if (sched_state.groups.has(group_name)) {
+                for (const auto& child_gname : sched_state.groups(group_name).groups()) {
+                    const auto gefac = sched_state.groups(child_gname).getGroupEfficiencyFactor();
+                    has_values = has_values || self(self, child_gname, rates, gefac);
```

Comment:
Potential issue here: If `has_values` is true, e.g., from the first child, then we won't call `self()` for subsequent children because of short-circuiting.  Is that intentional?  If so, please add a comment for why that's the intended semantics.  If not, maybe we could rewrite it as
```C++
const auto child_has_values = self(...);
has_values = has_values || child_has_values;
```

Reply:
Oops, you're right, that's a real issue.. thanks! Sufficient to change the order though, right? (I.e., `has_values = self(self, child_gname, rates, gefac) || has_values; `)